### PR TITLE
#1352: refactor voting handler

### PR DESCRIPTION
**Pull request by rz-p**
**WIP**
Only moved the functions to `VotingHandler`

### CODE REVIEW

**Comment by rz-p** [comment]
**BUG**
- Voting button doesn't work
- Admin doesn't have the CardButton shown

File: hms-app/src/components/card-details/VotingHandler.tsx [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+            autoClose: 2000,
+          })
+        }
+      }
+    )
+  }
+
+  const removeThisVote = () => {
+    removeVote(removeIdeaVoteParticipant, setVoteCheck)
+  }
```
Comment (rz-p) [inline hms-app/src/components/card-details/VotingHandler.tsx:105]: function to call the removeVote

File: hms-app/src/components/card-details/VotingHandler.tsx [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+      <Button
+        disabled={buttonIsDisabled}
+        onClick={voteCheck ? removeThisVote : addThisVote}
+        style={{
+          backgroundColor: voteCheck ? LEAVE_BUTTON_COLOR : JOIN_BUTTON_COLOR,
+        }}
+      >
+        {voteCheck ? 'Remove Vote' : 'Add Vote'}
+      </Button>
+    </Card.Section>
```
Comment (rz-p) [inline hms-app/src/components/card-details/VotingHandler.tsx:204]: **BUG**
- Counter not updating automatically after adding vote
- Button stays blue after voting and refreshing page. Turns red only after clicking

File: hms-app/src/components/card-details/IdeaDetails.tsx [hunk 3]
```diff
@@ -45,8 +24,10 @@ import { getCategoryDetails } from '../../actions/CategoryActions'
 import { getSkillDetails } from '../../actions/SkillActions'
 import IdeaCommentDetails from './IdeaCommentDetails'
 import CardButton from './CardButton'
+import ParticipantsHandler from './ParticipantsHandler'
+import VotingHandler from './VotingHandler'
 
-type IProps = {
+export type IProps = {
```
Comment (LetsPoke) [inline hms-app/src/components/card-details/IdeaDetails.tsx:30]: why?

File: hms-app/src/components/card-details/IdeaDetails.tsx [hunk 4]
```diff
@@ -56,33 +37,21 @@ export default function IdeaDetails(props: IProps) {
...
   const [loader, setLoader] = useState(false)
   const [ideaData, setIdeaData] = useState(idea)
-
-  const loadIdeaData = () => {
-    getIdeaDetails(instance, ideaData.id).then((data) => {
-      setIdeaData(data)
-      setLoader(false)
-    })
-  }
+  const [participantInfo, setParticipantInfo] = useState({
```
Comment (LetsPoke) [inline hms-app/src/components/card-details/IdeaDetails.tsx:51]: looks like participantInfo is never used

**Review by LetsPoke — APPROVED** [review]

**Pull request merged**